sound/soc/sof/ipc4-topology.c

```diff
@@ -733,6 +735,10 @@ static int sof_ipc4_widget_setup_comp_pga(struct snd_sof_widget *swidget)
 
 	sof_ipc4_widget_update_kcontrol_module_id(swidget);
 
+	for (i = 0; i < swidget->num_tuples; i++ )
+		if (swidget->tuples[i].token == SOF_TKN_GAIN_USE_TOPOLOGY_NAME)
+			swidget->widget->no_wname_in_long_name = swidget->tuples[i].value.v;
```

Comment:
I am really confused here.

Is this the case that we provide "a full control over the user visible mixer name" ?

The example just seems to drop the prefix, that's hardly full control.

Reply:
We already had the full control except for the prefix, that is always added, if the newly added flag is not set. The rest of the name comes from topology2 mixer object's "name"-attribute:

https://github.com/thesofproject/sof/blob/45c3f132151676b77d7c290860ddd6b0b2abcdf7/tools/topology/topology2/cavs-mixin-mixout-hda.conf#L43

The point of this series is to provide means to get rid of the non sense widget name e.g. "gain.2.1" in the beginning of the user visible mixer name.

Reply:
suggested edit to the commit message

Add SOF_TKN_GAIN_NAME_NO_PREFIX token, sof_topology_token entry in
gain_tokens array, and the associated code in sof_ipc4_widget_setup_comp_pga()
to set no_wname_in_long_name flag in the associated snd_soc_dapm_widget.

In practice setting the "name_no_prefix" attribute in the gain node
means that the widget name is not added as a
prefix to the mixer name.

E.g. "gain.2.1 Post Mixer Analog Playback Volume" becomes just
"Post Mixer Analog Playback Volume".

The decision to skip the prefix addition is made when the topology binary file is created. The topology can be modified on a need-basis to make the control names more user-friendly.

Reply:
this part needs to go to sof_widget_ready()

Reply:
nothing seems to have changed here either, so this is probably not quite ready for a merge.